IHP/IDE/Defaults/TableColumnDefaults.hs

```diff
@@ -0,0 +1,425 @@
+@
+
+-}
+defCreateTableWSetCol :: Text -> Text -> PostgresType -> CreateTable
+defCreateTableWSetCol tablename columnname pgt = defCreateTableWCol tablename (pure $ setColumn columnname pgt)
+
+{- | Same as its progenitor `defCreateTableWSetCol` except it uses `setColumnN`
+-}
+defCreateTableWSetColN :: Text -> Text -> PostgresType -> CreateTable
+defCreateTableWSetColN tablename columnname pgt = defCreateTableWCol tablename (pure $ setColumnN columnname pgt)
+
+
+{- | Takes the name of the table, the items you want inside the primaryKeyConstraint and a list of columns
+to return a table where the primary key constraint is set.
+
+__Example:__ 
+
+@ 
+let orderTrucksTable = defCreateTablePKID "orderTrucks" ["order_id","truck_id"] cols
+                  where cols = map mkColumn ["order_id","truck_id"]
+                        mkColumn x = (setColumnN x PBigserial)
+@
+
+>>> orderTrucksTable
+CreateTable { name = "orderTrucks"
+            , columns = [ Column { name = "order_id", 
+                                   columnType = PBigserial, 
+                                   defaultValue = Nothing, 
+                                   notNull = True, 
+                                   isUnique = False, 
+                                   generator = Nothing}
+                        , Column { name = "truck_id", 
+                                   columnType = PBigserial, 
+                                   defaultValue = Nothing, 
+                                   notNull = True, 
+                                   isUnique = False, 
+                                   generator = Nothing}
+                         ]
+            , primaryKeyConstraint = PrimaryKeyConstraint {primaryKeyColumnNames = ["order_id","truck_id"]}
+            , constraints = []
+            , unlogged = False}
+-}
+defCreateTablePKID :: Text -> [Text] -> [Column] -> CreateTable
+defCreateTablePKID name items cols = (defCreateTableWCol name cols) {primaryKeyConstraint = PrimaryKeyConstraint items}
+
+
+{- | Allows you to set the name and columnType. Uses `defColumn` as its base
+
+If other values need to be changed, this can be done using: 
+@(setColumn a b){..}@
+
+__Example:__
+
+>>> setColumn "user_id" PTrigger 
+Column {name = "user_id", columnType = PTrigger, defaultValue = Nothing, notNull = False, isUnique = False, generator = Nothing}
+
+-}
+setColumn :: Text -> PostgresType -> Column
+setColumn name pgt = defColumn { name = name
+                               , columnType = pgt
+                               }
+
+-- | A version of `setColumn` where @notNull = True@
+setColumnN :: Text -> PostgresType -> Column
+setColumnN n p = (setColumn n p) {notNull = True}
+
+-- | Sets a column to have a default value. Would recommend using in conjunction with `setColumn`
+setColumnDefaultVal :: Maybe Expression -> Column -> Column
+setColumnDefaultVal expression column = column {defaultValue = expression}
+
+
```

Comment:
Functions below this lines are IMO not a good idea. By having these very specific functions outside of the test suites we actually make tests harder to change, because you now need to look at two files to understands what's going on. So it would be better to undo the changes below and just make use of `defTable` and `defColumn` inside the tests.

Reply:
So my thinking here was that if you had repeating units or tests, you can define them once, remember them and then use them as normal and so long as you know the definition of the table or column, you could:
1. reuse it
2. use it as a new base for a table/column

Perhaps it's not useful for whole tables but certainly default values/columns that may repeat and have complicated structures.

That said, I do see your point and there are several items that are superfluous; secondly, I'm sure if you needed to define repeatable units, you would define them as needed and perhaps on the test page like was done before.

Therefore, I will revert the tests to defintions of `defColumn` etc and leave space for IHP to define its own monomers as required.

Reply:
I was also thinking that the writer would check the definition of the test by ctrl+clicking on the name and opening up the references window -> but I guess not everyone uses vscode for IDE which is understandable.